### PR TITLE
FED - highlight current section/item in main nav

### PR DESCRIPTION
**Pull request by liamjohnston**
Examples:

Screenshot 2024-07-05 at 12 34 45 PM
Screenshot 2024-07-05 at 12 34 55 PM

### CODE REVIEW

File: cdhweb/static_src/components/MainNavDesktop/MainNavDesktop.tsx [hunk 1]
```diff
@@ -22,6 +24,7 @@ const DesktopMenu = ({
       overview: 'Search our website for people, projects, events or blogs.',
       l2_items: [],
       link_url: searchUrl,
+      is_current: isSearchPage,
```
Comment (liamjohnston) [inline cdhweb/static_src/components/MainNavDesktop/MainNavDesktop.tsx:27]: The search nav item isn't part of the json blob, it's hard-coded here. Hence the need to pass in whether the current page is the search page when this component mounts.

File: templates/includes/header.html [hunk 2]
```diff
@@ -48,7 +53,12 @@
                         </li>
                     {% endfor %}
                     <li>
-                        <a href="{% url 'search' %}" class="main-nav-desktop__item">
+                        <a href="{% url 'search' %}" 
+                            class="main-nav-desktop__item
+                            {% if request.path|starts_with:'/search/' %}
+                                main-nav-desktop__item--current-section
+                            {% endif %}
+                        ">
```
Comment (liamjohnston) [inline templates/includes/header.html]: This bit is a little dodgy. 

@sarahframe / @haydngreatnews Is there a way for me to check against that `{% url 'search' %}` variable? I couldn't figure it out, so I went with a comparing it to a hard-coded string of "/search/".

Note this is only for the pre-/non-JS render of the nav. It gets updated once react mounts.

Comment (sarahframe) [reply templates/includes/header.html]: Is it right that the only time search should be highlighted in the nav is when we are on the actual search page, as there won't be any child pages of search?

Comment (sarahframe) [reply templates/includes/header.html]: I've pushed a fix for this :) @liamjohnston

Comment (liamjohnston) [reply templates/includes/header.html]: That's correct. Thanks :)

File: cdhweb/static_src/components/MainNavDesktop/MainNavDesktopItem.tsx [hunk 3]
```diff
@@ -154,7 +153,12 @@
           )}
         </>
       ) : (
-        <a href={item.link_url} className="main-nav-desktop__item">
+        <a
+          href={item.link_url}
```
[the remark this answers is not in the document]
Comment (cameralibre) [reply cdhweb/static_src/components/MainNavDesktop/MainNavDesktopItem.tsx:157]: what are you talking about, github?

**Review by cameralibre — APPROVED** [review]
looks good to me, despite github's whining

File: cdhweb/static_src/components/MainNavDesktop/MainNavDesktopItem.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (liamjohnston) [reply cdhweb/static_src/components/MainNavDesktop/MainNavDesktopItem.tsx:157]: 🤷

**Pull request merged**